system/jlib/jsocket.hpp

```diff
@@ -331,6 +331,9 @@ class jlib_decl ISocket : extends IInterface
     // Get peer ip of socket - in UDP returns return addr
     virtual IpAddress &getPeerAddress(IpAddress &addr)=0;
 
+    // Get local endpoint of socket
+    virtual SocketEndpoint &getEndpoint(SocketEndpoint &ep)=0;
```

Comment:
trivial: could be const method, but other existing methods (e.g. getPeerAddress) should be too.